rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/EvaluatorUtil.java

```diff
@@ -80,34 +85,23 @@ public static <T> InterruptibleFuture<T> runEvaluator(String task, CompletableFu
                         return defaultResult;
                     } finally {
                         actualEval.jobEnd(task, jobSuccess);
-                        actualEval.__setInterrupt(false);
                         runningEvaluator.set(null);
+                        actualEval.__setInterrupt(false);
                     }
                 }
             }
             catch (Throw e) {
                 logger.error("Internal error during {}\n{}: {}\n{}", task, e.getLocation(), e.getMessage(),
                         e.getTrace());
-                // logger.error("Full internal error: ", e);
-                if (throwFailure) {
-                    throw e;
-                }
-                return defaultResult;
+                throw new ResponseErrorException(new ResponseError(ResponseErrorCode.RequestFailed, formatMessage((Throw)e), null));
```

Comment:
Sonar reports an unnecessary cast here

Reply:
Thanks, after refactoring I forgot to check these messages.